### PR TITLE
Suppress analytics flag pass through to analysis server

### PR DESCRIPTION
**Pull request by jcollins-g**
This allows `flutter analyze` to pass through the new `--suppress-analytics` flag in analysis server when `--suppress-analytics` is on the `flutter` command line.

Issue: https://github.com/dart-lang/sdk/issues/49445

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by andrewkolos — APPROVED** [review]
LGTM; but, if this change can wait, I'd consider getting a review from @eliasyishak or @christopherfujino since they are likely more familiar with the context of this change

**Comment by christopherfujino** [comment]
@jcollins-g is this part of the unified analytics project? That is, relying on https://pub.dev/packages/unified_analytics. I thought that the idea was that the analyzer would have direct access to whether or not the user has opted in to analytics?

**Comment by jcollins-g** [comment]
> @jcollins-g is this part of the unified analytics project? That is, relying on https://pub.dev/packages/unified_analytics. I thought that the idea was that the analyzer would have direct access to whether or not the user has opted in to analytics?

Yes, and you are correct in the common case.  This change is for the edge case when a flutter user is opted in to analytics generally, but has passed in `--suppress-analytics` on the command line for a one-time disabling of analytics and expects that to apply to `analyze` as well as other Flutter tools.

**Comment by christopherfujino** [comment]
> > @jcollins-g is this part of the unified analytics project? That is, relying on https://pub.dev/packages/unified_analytics. I thought that the idea was that the analyzer would have direct access to whether or not the user has opted in to analytics?
> 
> Yes, and you are correct in the common case. This change is for the edge case when a flutter user is opted in to analytics generally, but has passed in `--suppress-analytics` on the command line for a one-time disabling of analytics and expects that to apply to `analyze` as well as other Flutter tools.

Ahh, ok, TIL the flutter tool implements that flag :) SGTM

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/executable.dart
Comment (jcollins-g) [reply packages/flutter_tools/lib/executable.dart]: done.

**Comment by christopherfujino** [comment]
@jcollins-g any time you're ready to merge this, you can add the "autosubmit" label to this PR, and a bot will merge it once the tree is green.

**Pull request merged**
